Foundation/include/Poco/NumberFormatter.h

```diff
@@ -820,17 +1055,33 @@ inline std::string NumberFormatter::format0(UInt64 value, int width)
 
 
 inline std::string NumberFormatter::formatHex(UInt64 value, bool prefix)
+{
+	return formatHex(value, prefix ? Options::OPT_PREFIX : Options::OPT_NONE);
+}
+
+
+inline std::string NumberFormatter::formatHex(UInt64 value, Options options)
 {
 	std::string result;
-	intToStr(value, 0x10, result, prefix);
+	const bool prefix { (options & Options::OPT_PREFIX) != Options::OPT_NONE};
+	const bool lowercase { (options & Options::OPT_LOWERCASE) != Options::OPT_NONE};
+	intToStr(value, 0x10, result, prefix, -1, ' ', 0, lowercase);
 	return result;
 }
 
 
 inline std::string NumberFormatter::formatHex(UInt64 value, int width, bool prefix)
+{
+	return formatHex(value, width, prefix ? Options::OPT_PREFIX : Options::OPT_NONE);
+}
+
+
+inline std::string NumberFormatter::formatHex(UInt64 value, int width, Options options)
 {
 	std::string result;
-	intToStr(value, 0x10, result, prefix, width, '0');
+	const bool prefix { (options & Options::OPT_PREFIX) != Options::OPT_NONE};
+	const bool lowercase { (options & Options::OPT_LOWERCASE) != Options::OPT_NONE};
+	intToStr(value, 0x10, result, prefix, width, '0', 0, lowercase);
```

Comment:
I'm totally ok that these are in header with inline as these really get optimized away basically. How ever I wonder should we move deprecated to cpp file. We do not care those so much as they are really deprecated. This heder get's quite big because of those.

In the end totally your call.

Reply:
yes, these should probably go to cpp file. a rule of thumb is single liners and templates in headers, everything else in cpp

Reply:
Moved deprecated functions to cpp file.